build/gulp/renovation-testing-playground.js

```diff
@@ -0,0 +1,182 @@
+                .pipe(gulp.dest(path.dirname(fileName))), `${platform}-declaration-app`);
+        };
+        const generateDeclarationFiles = platformDeclarationFiles.map(declarationAppFileGeneratorTask);
+        const compileSourceGeneratorTask = (fileEntries, dest) => {
+            const getConfig = () => {
+                const webpackConfig = require(path.resolve(path.join(platformRoot, './webpack.config')))(path.resolve(platformRoot));
+                const entries = fileEntries
+                .reduce((result, fn) => {
+                    result[entryName(fn)] = path.resolve(getEntyPoint(fn))
+                    return result;
+                }, {});
+                const _config = {
+                    ...webpackConfig,
+                    ...{
+                        watch: isWatch,
+                        entry: entries,
+                        output: {
+                            path: path.resolve(dest),
+                            publicPath: '/',
+                            filename: '[name].js',
+                            chunkFilename: '[id].chunk.js',
+                        },
+                        resolveLoader: {
+                            modules: [path.resolve('testing/renovation/node_modules'), 'node_modules'],
+                        },
+                    },
+                };
+                return _config;
+            };
+            // use lazypipe to prevent resolve webpack require dependencies
+            const webpackPipe = (config) => lazypipe().pipe(webpackStream, config, webpack)();
+            return namedTask(() => gulp
+                .src(path.resolve(platformRootSrc))
+                .pipe(webpackPipe(getConfig()))
+                .pipe(gulp.dest(dest)), `${platform}-compile-source`);
+        }
+        const compileSourceFiles = [
+            compileSourceGeneratorTask(nativeFiles, platformRootDest),
+            compileSourceGeneratorTask(platformDeclarationFiles, platformDeclarationDest)
+        ]
+        return gulp.series(
+            gulp.parallel([...generateHtmlFiles, ...generateDeclarationFiles]),
+            gulp.parallel(compileSourceFiles)
+        );
+    });
+
+gulp.task('build-renovation-testing', gulp.series(
+    (done) => { process.env.BABEL_ENV = 'development'; done(); },
+    gulp.parallel(tasks({ iswatch: false }))
+));
+
+gulp.task('build-renovation-testing:watch', gulp.series(
+    (done) => { process.env.BABEL_ENV = 'development'; done(); },
+    gulp.parallel(tasks({ iswatch: true }))
+));
+
+const foldersToCleanup = [
+    ...(glob.sync(renovationRoot + '/**/dist')),
+    ...(glob.sync(renovationRoot + '/**/src/declaration')),
+    ...(glob.sync(renovationRoot + '/vue/src/**/*-app.js')),
+]
+    .map(f => {
+        return () => {
+            return del(f)
+        }
+    });
+gulp.task('clean-renovation-testing',
+    gulp.parallel([...foldersToCleanup, (cb) => { cb(); }
+    ])
+);
```

Comment:
No new line here and in several more places. Don't you use VSCode Workspace?

Reply:
I use workspaces.